tests/test_aci_worker.py

```diff
@@ -0,0 +1,942 @@
+
+    aci_job_config = await create_job_configuration(
+        aci_credentials, worker_flow_run, {"command": command}
+    )
+
+    assert aci_job_config.command == command
+
+
+def test_worker_invalid_command_validation(aci_credentials):
+    # ensure invalid commands cause a validation error
+    with pytest.raises(ValueError):
+        AzureContainerJobConfiguration(
+            command=["invalid_command", "arg1", "arg2"],  # noqa
+            subscription_id=SecretStr("test"),
+            resource_group_name="test",
+            aci_credentials=aci_credentials,
+        )
+
+
+async def test_worker_container_client_creation(
+    aci_worker, worker_flow_run, job_configuration, aci_credentials, monkeypatch
+):
+    # verify that the Azure Container Instances client and Azure Resource clients
+    # are created correctly.
+
+    mock_azure_credential = Mock(spec=ClientSecretCredential)
+    monkeypatch.setattr(
+        prefect_azure.credentials,
+        "ClientSecretCredential",
+        Mock(return_value=mock_azure_credential),
+    )
+
+    # don't use the mock_aci_client or mock_resource_client_fixtures, because we want to
+    # test the call to the client constructors to ensure the block is calling them
+    # with the correct information.
+    mock_container_client_constructor = Mock()
+    monkeypatch.setattr(
+        prefect_azure.credentials,
+        "ContainerInstanceManagementClient",
+        mock_container_client_constructor,
+    )
+
+    mock_resource_client_constructor = Mock()
+    monkeypatch.setattr(
+        prefect_azure.credentials,
+        "ResourceManagementClient",
+        mock_resource_client_constructor,
+    )
+
+    subscription_id = "test_subscription"
+    job_configuration.subscription_id = SecretStr(value=subscription_id)
+    with pytest.raises(RuntimeError):
+        await aci_worker.run(worker_flow_run, job_configuration)
+
+    mock_resource_client_constructor.assert_called_once_with(
+        credential=mock_azure_credential,
+        subscription_id=subscription_id,
+    )
+    mock_container_client_constructor.assert_called_once_with(
+        credential=mock_azure_credential,
+        subscription_id=subscription_id,
+    )
+
+
+@pytest.mark.usefixtures("mock_aci_client")
+async def test_worker_credentials_are_used(
+    aci_worker,
+    worker_flow_run,
+    job_configuration,
+    aci_credentials,
+    mock_aci_client,
+    mock_resource_client,
+    monkeypatch,
+):
+    (client_id, client_secret, tenant_id) = credential_values(aci_credentials)
+
+    mock_client_secret = Mock(name="Mock client secret", return_value=client_secret)
+    mock_credential = Mock(wraps=ClientSecretCredential, return_value=Mock())
+
+    monkeypatch.setattr(
+        aci_credentials.client_secret, "get_secret_value", mock_client_secret
+    )
+    monkeypatch.setattr(
+        prefect_azure.credentials, "ClientSecretCredential", mock_credential
+    )
+
+    with pytest.raises(RuntimeError):
```

Comment:
Why is the worker expected to raise a runtime error here?

Reply:
See previous reply